### PR TITLE
[Update Detekt Configuration] Build Upon Default Config & Enable All Modules

### LINKED ISSUE #16893: Update Detekt Configuration - Build Upon Default Config
Parent #16892

This issue is about changing Detekt's `detekt.yml` configuration so that it is being build upon the default config (`buildUponDefaultConfig = true`, just like it is already done on WCAndroid.

### PR DESCRIPTION
**Pull request by ParaskP7**
Parent: #16892
Closes: #16893

This PR mainly:
1. Enables Detekt to build upon default configuration (cf55e492c845fa8e3b2a74610adab10058110278).
2. Enabled Detekt by default for all modules (04ca21f1898d5c475fe710c4188c995bd32c1ef6).
3. Resolves/Suppresses warnings on all modules and sources (c263b14aad18d0d33c437cad6c9fd1559a3c0ad1 + 1d7b0f6f78e0d49f8c33bbb7928b75fc7107d873 + 042321e080bd1463297045b6c9ea5d1ed7cd6fcf + 21ce4e9033d19f5ac730e9f3a1c97a95eb869a04 + 4d6a2dadb2720262df3fc47d9b28859a00332fe2 + dfffd933ef1a929aefac07598c9d1c0f08cf24f4 + 9e01c27da7892c00a4cede369286d84a38687876 + 21f525387a48e0b94d7ff36d8f2827b0cc8c85ea).
4. Updates Detekt baseline (42e7428f0401057b0b9336b65b61089bbc24fe1a + bb0358696bf0ae3ddb971eee7676a5b32fbe674d).
5. Updates CI to respect all modules (f26147b7a2b2c48cc3a3b55a83d1cd769013deb1).
6. Updates Docs to respect all modules (17da9689a9fcb0d1f43451f947133241d3afc960).
7. Enable building Detekt in parallel since we are now using Detekt on multiple modules (aa9bf645d20418b03abb4c8d65b06335660bab09).

Enabling building upon default configuration is done in order to make it easier to update Detekt going forward and at the same time making `detekt.yml` file less complex to traverse (in terms of updated rules and comparing that to the default).

-----

To test:

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough (and more specifically, the [detekt](https://buildkite.com/automattic/wordpress-android/builds/5568#01821626-e96f-4481-9ef8-178af200a584) one).
- However, if you really want to be thorough you can do the following:
    1. `Build Upon Default Config`
        - Run `./gradlew detekt` and make sure it successfully completes (🟢).
        - Comment out the `MagicNumber` rule from within the `detekt.yml` file and run `./gradlew detekt` again. You will now notice that Detekt reports new warnings (⚠️).
        - Add back the `MagicNumber` rule from within the `detekt.yml` file and run `./gradlew detekt` again. Detekt should be 🟢 again.
    2. `Enable All Modules`
        - Run `./gradlew detekt` and make sure it successfully completes (🟢).
        - Go to a `libs` related module, for example the `image-editor`, artificially add a new Detekt warning, or removed a suppressed warning (for example the `TooManyFunctions`) and run `./gradlew detekt` again. You will now notice that Detekt reports new warnings (⚠️), specific to this module.
        - Remove this warning that you added artificially, or add back the suppress warnings that you removed, and run `./gradlew detekt` again. Detekt should be 🟢 again.

-----

## Regression Notes
1. Potential unintended areas of impact

Detekt is not working as expected, skipping some rules that were previously enabled explicitly (see `formatting` for example).

8. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

9. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by ovitrif — APPROVED** [review]
Great work @ParaskP7!

The changes look good to me, I've run all testing steps and everything works as expected 🎉 

Thank you for this work 🙇 🥇 🚀 💯 

Leaving this unmerged in case you're waiting for more feedback from the @wordpress-mobile/owl-team, though the changes are quite straightforward imho.
Feel free to merge this if you need it merged sooner in order to unblock the next steps 🙋🏻‍♂️

**Comment by ParaskP7** [comment]
Thank you so much for reviewing and testing this PR @ovitrif , you rock! 🙇 ❤️ 🚀

**Pull request merged**
